build.gradle

```diff
@@ -66,8 +67,42 @@ final testNGVersion = '6.11'
 
 final baseJarName = 'gatk'
 final secondaryBaseJarName = 'hellbender'
-
 final pythonPackageArchiveName = 'gatkPythonPackageArchive.zip'
+final largeResourcesFolder = "src/main/resources/large"
+final lfsFileSentinel = "version https://git-lfs.github.com/spec/v1"
+final buildPrerequisitesMessage = "See https://github.com/broadinstitute/gatk#building for information on how to build GATK."
+
+// Large runtime resource files must be present at build time to be compiled into the jar, so validate
+// that the lfs files have been fetched and that the directory does not contain the git-lfs pointer files
+def validateLargeResourceFile(largeResourceFile, lfsSentinel, buildPrerequisitesMessage) {
+    def readBytesFromFile = { largeFile, n ->
+        final byte[] bytes = new byte[n]
+        largeResourceFile.withInputStream { stream -> stream.read(bytes, 0, bytes.length) }
+        return bytes
+    }
+    final byte[] actualBytes = readBytesFromFile(largeResourceFile, lfsSentinel.length());
+    if (new String(actualBytes, "UTF-8") == lfsSentinel) {
+        throw new GradleException(
+                "$largeResourceFile appears to be a git-lfs pointer file." +
+                        " The resource files in src/main/resources/large must be downloaded using git-lfs. " +
+                        " $buildPrerequisitesMessage")
+    }
+}
+
+// Validate that we have a clone of the repository and that any required git-lfs files have been resolved
+def validateBuildPrerequisites(javaVersion, largeResourcesFolder, lfsSentinel, buildPrerequisitesMessage) {
+    if (ToolProvider.getSystemToolClassLoader() == null) {
+        throw new GradleException("The SystemToolClassLoader obtained from the ToolProvider is null. A Java $RequiredJavaVersion JDK must be installed. $buildPrerequisitesMessage")
```

Comment:
I think it happens when you have a JRE rather than a JDK (see https://github.com/broadinstitute/gatk/issues/4532). Also, although I haven't tried it I suspect it will happen with JDK9.